.github/workflows/build-ruby.yml

```diff
@@ -0,0 +1,128 @@
+name: Build Ruby
+
+on:
+  workflow_dispatch:
+    inputs:
+      push:
+        description: Push images
+        required: true
+        type: boolean
+        default: true
+  push:
+    branches:
+      - "**"
```

Comment:
I assume we don't want to keep `push` as one of the triggers, right? I'd think we only want manual (`workflow_dispatch`).

Reply:
I do want to keep the `push` trigger.

> it's getting somewhat confusing to find the ones that fail in the github UI

It's mostly confusing because CircleCI does not show up in the `Checks` tab of a PR so we have to check the summary :/ (I just jump to a tab to CircleCI to really check, much easier)

I do wish GitHub would sort jobs by state: (in the order: failed, cancelled, success, running, pending)

That said, since these images are essentially independent of dd-trace-rb and [can be reused to great effect](https://github.com/DataDog/dd-trace-rb/pull/2380#discussion_r1029203607), I initially wanted these images to be defined and built outside of this repo, but @delner was not too keen about that.